### PR TITLE
search page filtering

### PR DESCRIPTION
**Pull request by Alexcchip**
Created filter functionality (profile button is there but no BE/FE work done yet). Also this is a new branch btw. Old one was scrapped (no one else was using it). I do plan on using this branch for the profile stuff after this though.
## Description

[Link to Ticket](https://github.com/GenerateNU/platnm/issues/129) and [Link to ticket](https://github.com/GenerateNU/platnm/issues/152)

## How Has This Been Tested?

Screenshots of tabs (not sure if the results/search endpoint works 100% though).

### CODE REVIEW

**Comment by Alexcchip** [comment]
image

image

image

**Comment by Alexcchip** [comment]
Also the CSS/layout of everything isnt done yet but I plan on finalizing that after profile is done.

File: backend/internal/storage/postgres/schema/user.go [hunk 1]
```diff
@@ -178,14 +206,22 @@ func (r *UserRepository) UpdateUserOnboard(ctx context.Context, email string, en
...
+		   COUNT(DISTINCT followers.follower_id) AS follower_count, 
+		   COUNT(DISTINCT followed.followee_id) AS followed_count
+	FROM "user" u
+	LEFT JOIN follower followers ON followers.followee_id = u.id
+	LEFT JOIN follower followed ON followed.follower_id = u.id
+	WHERE u.%s = $1
+	GROUP BY u.id, u.username, u.display_name, u.profile_picture, u.bio;
+`, idType)
 
 	exists, err := r.UserExists(ctx, id.String())
```
Comment (garrettladley) [inline backend/internal/storage/postgres/schema/user.go:226]: why have this query? seems unnecessary if your original query will fail when the user does not exist. seems to be an unnecessary round trip to the DB

File: backend/internal/storage/postgres/schema/user.go [hunk 2]
```diff
@@ -178,14 +206,22 @@ func (r *UserRepository) UpdateUserOnboard(ctx context.Context, email string, en
 	return enthusiasm, nil
 }
 
-func (r *UserRepository) GetUserProfile(ctx context.Context, id uuid.UUID) (*models.Profile, error) {
+func (r *UserRepository) GetUserProfile(ctx context.Context, id uuid.UUID, idType string) (*models.Profile, error) {
```
Comment (garrettladley) [inline backend/internal/storage/postgres/schema/user.go:209]: validate `idType` prior to use? or let query result handle invalid?

File: backend/internal/service/handler/users/user.go [hunk 3]
```diff
@@ -110,13 +122,22 @@ func (h *Handler) GetUserProfile(c *fiber.Ctx) error {
 		return errs.NotFound("User", "userID", id)
 	}
 
-	userUUID, err := uuid.Parse(id)
-	if err != nil {
-		print(err.Error())
-		return err
-	}
+	if flag == "id" {
```
Comment (garrettladley) [inline backend/internal/service/handler/users/user.go:125]: what is the purpose of this branch? have nesting when it makes sense, if this is the main path, why add an extra indentation?

File: backend/internal/service/handler/users/user.go [hunk 4]
```diff
@@ -99,7 +111,7 @@ func (h *Handler) CalculateScore(c *fiber.Ctx) error {
 
 func (h *Handler) GetUserProfile(c *fiber.Ctx) error {
```
Comment (garrettladley) [inline backend/internal/service/handler/users/user.go:112]: seems like a code smell if you switch on flag and have different underlying implementations. to me, this suggests they should be separate endpoints.

**Review by garrettladley — CHANGES_REQUESTED** [review]

File: backend/internal/storage/postgres/schema/user.go (hunk 2 above)
Comment (Alexcchip) [reply backend/internal/storage/postgres/schema/user.go:209]: I realized that I should likely just have 2 endpoints -- one for querying profile via ID and another that combines user/display names. It also went over my head that I wanted a list of profiles that are similar or equal to the parameter and not just a single profile. Will commit refactor asap.

File: backend/internal/service/handler/users/user.go (hunk 4 above)
Comment (Alexcchip) [reply backend/internal/service/handler/users/user.go]: yes spoke to Dessy about this yesterday and decided to get rid of flag

**Review by ddusichka — APPROVED** [review]
looks good! going to merge this now since other tickets depend on it but feel free to make a follow up PR with your styling changes

**Pull request merged**